src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/ios/LoginPage.java

```diff
@@ -63,19 +65,70 @@ public void checkPrivacyPolicyCheckbox() {
 		privacyPolicyCheckbox.click();
 	}
 
+	@Override
+	public boolean isPageOpened() {
+		return loginBtn.isElementPresent();
+	}
+
 	@Override
 	public CarinaDescriptionPageBase clickLoginBtn() {
 		loginBtn.click();
 		return initPage(getDriver(), CarinaDescriptionPageBase.class);
 	}
 
+	@Override
+	public WebViewPageBase clickLoginButton() {
+		loginBtn.click();
+		return initPage(getDriver(), WebViewPageBase.class);
+	}
+
 	@Override
 	public boolean isLoginBtnActive() {
 		return Boolean.parseBoolean(loginBtn.getAttribute("enabled"));
 	}
 
+
+
+
+	@Override
+	public boolean verifyFieldsArePresented() {
+		return passwordInputField.isElementPresent();
+	}
+
+	@Override
+	public boolean isMaleRadioBtnUnchecked() {
+		return false;
+	}
+
+
+	@Override
+	public boolean isFemaleRadioBtnChecked() {
+		return femaleRadioBtn.isChecked();
+	}
+
+	@Override
+	public boolean isPrivacyPolicyCheckboxUnchecked() {
+		privacyPolicyCheckbox.uncheck();
+		return false;
+	}
+
+	@Override
+	public boolean isPrivacyPolicyCheckboxChecked() {
+		return privacyPolicyCheckbox.isChecked();
+	}
+
+	@Override
+	public String getTypeName() {
+		return nameInputField.getText();
+	}
+
+	@Override
+	public String getTypePassword() {
+		return passwordInputField.getText();
+	}
```

Comment:
if you don’t implement for ios use this
private static final String THIS_METHOD_IS_NOT_IMPLEMENTED_FOR_IOS = "This method is not yet implemented for iOS";
throw new UnsupportedOperationException(THIS_METHOD_IS_NOT_IMPLEMENTED_FOR_IOS);